subsys/net/ip/net_if.c

```diff
@@ -4732,6 +4733,147 @@ struct net_if *net_if_get_first_wifi(void)
 	return NULL;
 }
 
+int net_if_get_name(struct net_if *iface, char *buf, int len)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL || len <= 0) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(net_if_get_config(iface)->name);
+	if (name_len > len) {
+		return -ERANGE;
+	}
+
+	strncpy(buf, net_if_get_config(iface)->name, name_len);
+
+	return name_len;
+#else
+	return -ENOTSUP;
+#endif
+}
+
+int net_if_set_name(struct net_if *iface, char *buf)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	int name_len;
+
+	if (iface == NULL || buf == NULL) {
+		return -EINVAL;
+	}
+
+	name_len = strlen(buf);
+	if (name_len > CONFIG_NET_INTERFACE_NAME_LEN) {
+		return -ENAMETOOLONG;
+	}
+
+	memset(net_if_get_config(iface)->name, 0,
+	       CONFIG_NET_INTERFACE_NAME_LEN + 1);
+	strncpy(net_if_get_config(iface)->name, buf, name_len);
+
+	return 0;
+#else
+	return -ENOTSUP;
+#endif
+}
+
+int net_if_get_by_name(const char *name)
+{
+#if defined(CONFIG_NET_INTERFACE_NAME)
+	if (name == NULL) {
+		return -EINVAL;
+	}
+
+	STRUCT_SECTION_FOREACH(net_if, iface) {
+		if (strncmp(net_if_get_config(iface)->name, name, strlen(name)) == 0) {
+			return net_if_get_by_iface(iface);
+		}
+	}
+
+	return -ENOENT;
+#else
+	return -ENOTSUP;
+#endif
+}
+
+#if defined(CONFIG_NET_INTERFACE_NAME)
+static void set_default_name(struct net_if *iface)
+{
+	char name[CONFIG_NET_INTERFACE_NAME_LEN + 1];
+	int ret;
+
+#if defined(CONFIG_NET_L2_ETHERNET)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(ETHERNET)) {
+		static int count;
+
+		snprintk(name, sizeof(name) - 1, "eth%d", count++);
+	} else
+#endif /* CONFIG_NET_L2_ETHERNET */
+
+#if defined(CONFIG_NET_L2_IEEE802154)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(IEEE802154)) {
+		static int count;
+
+		snprintk(name, sizeof(name) - 1, "ieee%d", count++);
+	} else
+#endif /* CONFIG_NET_L2_IEEE802154 */
+
+#if defined(CONFIG_NET_L2_BLUETOOTH)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(BLUETOOTH)) {
+		static int count;
+
+		snprintk(name, sizeof(name) - 1, "bt%d", count++);
+	} else
+#endif /* CONFIG_NET_L2_BLUETOOTH */
+
+#if defined(CONFIG_NET_L2_DUMMY)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(DUMMY)) {
+		static int count;
+
+		snprintk(name, sizeof(name) - 1, "dummy%d", count++);
+	} else
+#endif /* CONFIG_NET_L2_DUMMY */
+
+#if defined(CONFIG_NET_L2_CANBUS_RAW)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(CANBUS_RAW)) {
+		static int count;
+
+		snprintk(name, sizeof(name) - 1, "can%d", count++);
+	} else
+#endif /* CONFIG_NET_L2_CANBUS_RAW */
+
+#if defined(CONFIG_NET_L2_PPP)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(PPP)) {
+		static int count;
+
+		snprintk(name, sizeof(name) - 1, "ppp%d", count++);
+	} else
+#endif /* CONFIG_NET_L2_PPP */
+
+#if defined(CONFIG_NET_L2_WIFI)
+	if (net_if_l2(iface) == &NET_L2_GET_NAME(WIFI)) {
```

Comment:
We don't have a proper "WIFI" L2 yet, so this won't work. Wi-Fi interfaces are exposed as "ETHERNET" with a flag set on an interface that indicate the interface is WiFi, there is a dedicated function to check that (`net_if_is_wifi()`).